tests/test_cognitoidp/test_cognitoidp.py

```diff
@@ -1164,13 +1374,31 @@ def test_admin_get_missing_user():
     caught.should.be.true
 
 
+@mock_cognitoidp
+def test_admin_get_missing_user_with_username_attributes():
+    conn = boto3.client("cognito-idp", "us-west-2")
+
+    username = "test@example.com"
+    user_pool_id = conn.create_user_pool(
+        PoolName=str(uuid.uuid4()), UsernameAttributes=["email"]
+    )["UserPool"]["Id"]
+
+    caught = False
```

Comment:
We should use the `with pytest.raises()` here, instead of manually verifying whether it has failed.
That way we can also verify whether the error message/code is correct